Mathlib/Topology/Category/CompHausLike/Limits.lean

```diff
@@ -0,0 +1,358 @@
+    (fun _ ↦ pullback.lift_snd _ _ _ _ _)
+    (fun _ _ hm ↦ pullback.hom_ext _ _ _ _ (hm .left) (hm .right))
+
+instance: HasLimit (cospan f g) where
+  exists_limit := ⟨⟨pullback.cone f g, pullback.isLimit f g⟩⟩
+
+/-- The functor to `TopCat` creates pullbacks if they exist. -/
+noncomputable instance : CreatesLimit (cospan f g) (compHausLikeToTop P) := by
+  refine createsLimitOfFullyFaithfulOfIso (pullback f g)
+    (((TopCat.pullbackConeIsLimit f g).conePointUniqueUpToIso
+        (limit.isLimit _)) ≪≫ Limits.lim.mapIso (?_ ≪≫ (diagramIsoCospan _).symm))
+  exact Iso.refl _
+
+/-- The functor to `TopCat` preserves pullbacks. -/
+noncomputable instance : PreservesLimit (cospan f g) (compHausLikeToTop P) :=
+  preservesLimitOfCreatesLimitAndHasLimit _ _
+
+/-- The functor to another `CompHausLike` preserves pullbacks. -/
+noncomputable instance {P' : TopCat → Prop}
+    (h : ∀ (X : CompHausLike P), P X.toTop → P' X.toTop) :
+    PreservesLimit (cospan f g) (toCompHausLike h) := by
+  have : PreservesLimit (cospan f g) (toCompHausLike h ⋙ compHausLikeToTop P') :=
+    inferInstanceAs (PreservesLimit _ (compHausLikeToTop _))
+  exact preservesLimitOfReflectsOfPreserves (toCompHausLike h) (compHausLikeToTop P')
+
+variable (P) in
+/--
+A typeclass describing the property that forming all explicit pullbacks is stable under the
+property `P`.
+-/
+class HasExplicitPullbacks : Prop where
+  hasProp {X Y B : CompHausLike P} (f : X ⟶ B) (g : Y ⟶ B) : HasExplicitPullback f g
+
+attribute [instance] HasExplicitPullbacks.hasProp
+
+instance [HasExplicitPullbacks P] : HasPullbacks (CompHausLike P) where
+  has_limit F := hasLimitOfIso (diagramIsoCospan F).symm
+
+variable (P) in
+/--
+A typeclass describing the property that explicit pullbacks along inclusion maps into disjoint
+unions is stable under the property `P`.
+-/
+class HasExplicitPullbacksOfInclusions [HasExplicitFiniteCoproducts.{0} P] : Prop where
+  hasProp : ∀ {X Y Z : CompHausLike P} (f : Z ⟶ X ⨿ Y), HasExplicitPullback coprod.inl f
+
+attribute [instance] HasExplicitPullbacksOfInclusions.hasProp
+
+instance [HasExplicitPullbacks P] [HasExplicitFiniteCoproducts.{0} P] :
+    HasExplicitPullbacksOfInclusions P where
+  hasProp _ := inferInstance
+
+end Pullbacks
+
+section FiniteCoproducts
+
+variable {P : TopCat.{u} → Prop} [HasExplicitFiniteCoproducts.{0} P]
+
+instance [HasExplicitPullbacksOfInclusions P] : HasPullbacksOfInclusions (CompHausLike P) where
+  hasPullbackInl _ := inferInstance
+
+theorem hasPullbacksOfInclusions
+    (hP' : ∀ ⦃X Y B : CompHausLike.{u} P⦄ (f : X ⟶ B) (g : Y ⟶ B)
+      (_ : OpenEmbedding f), HasExplicitPullback f g) :
+    HasExplicitPullbacksOfInclusions P :=
+  { hasProp := by
+      intro _ _ _ f
+      apply hP'
+      exact Sigma.openEmbedding_ι _ _ }
+
+/-- The functor to `TopCat` creates pullbacks of inclusions if they exist. -/
+noncomputable instance [HasExplicitPullbacksOfInclusions P] :
+    PreservesPullbacksOfInclusions (compHausLikeToTop P) :=
```

Comment:
The Lean statement doesn't seem to be about creating pullbacks, only preserving them.